From a107897fa935be493ba3d41ef072ec2560be376b Mon Sep 17 00:00:00 2001
From: Gaosheng Cui <cuigaosheng1@huawei.com>
Date: Tue, 18 Apr 2023 08:48:40 -0400
Subject: [PATCH 0083/1646] [Huawei] Revert "[Backport] crypto: api - Fix
 boot-up crash when crypto manager is disabled"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188696

--------------------------------

This reverts commit 937ed41d59712ab16996e0b3c70c82eaa8deaa0a.

The patch set of 937ed41d5971 ("crypto: api - Fix boot-up crash when
crypto manager is disabled") conflicts with the SIMD feature, and
maybe cause kernel panic on arm64, so revert the patch set.

Fixes: 937ed41d5971 (crypto: api - Fix boot-up crash when crypto manager is disabled)
Signed-off-by: Gaosheng Cui <cuigaosheng1@huawei.com>
---
 crypto/algapi.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/crypto/algapi.c b/crypto/algapi.c
index 0d0c3e937e36..f6481cb79946 100644
--- a/crypto/algapi.c
+++ b/crypto/algapi.c
@@ -284,8 +284,6 @@ static struct crypto_larval *__crypto_register_alg(struct crypto_alg *alg)
 
 	if (larval)
 		list_add(&larval->alg.cra_list, &crypto_alg_list);
-	else
-		alg->cra_flags |= CRYPTO_ALG_TESTED;
 
 	crypto_stats_init(alg);
 
-- 
2.17.1

